module/zfs/zio.c

```diff
@@ -5281,8 +5346,10 @@ zio_dva_throttle_done(zio_t *zio)
 	 * allocated the constituent blocks.  The first use their parent as tag.
 	 */
 	if (pio->io_child_type == ZIO_CHILD_GANG &&
-	    (pio->io_flags & ZIO_FLAG_IO_REWRITE))
+	    (pio->io_flags & ZIO_FLAG_IO_REWRITE)) {
 		tag = zio_unique_parent(pio);
+		size = SPA_OLD_GANGBLOCKSIZE;
```

Comment:
To clarify; this is just to match the call to `metaslab_group_alloc_increment()` inside `metaslab_alloc_range()`, which will always charge the requested psize to the throttle even if it allocates less?

(I mean, if it was throttled what else could it mean; it didn't _do_ an allocation).

It's fine, mostly just checking. You have seen my frothing internally at the interactions between the the zio and metaslab code, so "yeah, its fine, society could be improved somewhat" is all good :melting_face: